### PR TITLE
BROKERS: Select Post Impression By Ids

### LINKED ISSUE #244: BROKERS: Select Post Impression By Ids

### PR DESCRIPTION
**Pull request by evangelinedrink**
Closes #244

### CODE REVIEW

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs [hunk 1]
```diff
@@ -39,5 +40,18 @@ public async ValueTask<PostImpression> UpdatePostImpressionAsync(PostImpression
...
+            Guid PostId,
+            Guid ProfileId)
+        {
+            using var broker =
+                new StorageBroker(this.configuration);
+
+            broker.ChangeTracker.QueryTrackingBehavior =
+                QueryTrackingBehavior.NoTracking;
+
+            return await broker.PostImpressions.FindAsync(PostId, ProfileId);
```
Comment (ElbekDeveloper) [inline Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs:54]: Does EF know that this method needs to be queried based on these two `Ids`?

File: Taarafo.Core/Brokers/Storages/IStorageBroker.PostImpressions.cs [hunk 2]
```diff
@@ -12,5 +13,8 @@ public partial interface IStorageBroker
     {
         ValueTask<PostImpression> InsertPostImpressionAsync(PostImpression postImpression);
         ValueTask<PostImpression> UpdatePostImpressionAsync(PostImpression postImpression);
+        ValueTask<PostImpression> SelectPostImpressionByIdsAsync(
```
[the remark this answers is not in the document]
Comment (evangelinedrink) [reply Taarafo.Core/Brokers/Storages/IStorageBroker.PostImpressions.cs:15]: I have moved the SelectPostImpressionByIdsAsync above the UpdatePostImpressionAsync in the commit named Code Rub: Moved Select Post Impression By Ids.

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs (hunk 1 above)
Comment (evangelinedrink) [reply Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs:54]: I don't think the EF knows that the method needs to be queried based on the two Ids.  How would I be able to verify this?

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs [hunk 3]
```diff
@@ -27,6 +28,19 @@ public async ValueTask<PostImpression> InsertPostImpressionAsync(PostImpression
             return postImpressionEntityEntry.Entity;
         }
 
+        public async ValueTask<PostImpression> SelectPostImpressionByIdsAsync(
+            Guid PostId,
+            Guid ProfileId)
+        {
+            using var broker =
```
Comment (hassanhabib) [inline Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs:35]: Remove `using`

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs [hunk 4]
```diff
@@ -27,6 +28,19 @@ public async ValueTask<PostImpression> InsertPostImpressionAsync(PostImpression
...
+                new StorageBroker(this.configuration);
+
+            broker.ChangeTracker.QueryTrackingBehavior =
+                QueryTrackingBehavior.NoTracking;
```
Comment (hassanhabib) [inline Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs:39]: You don't need that anymore.

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs (hunk 3 above)
Comment (evangelinedrink) [reply Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs]: I was able to remove the using keywords in this file.

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs (hunk 4 above)
Comment (evangelinedrink) [reply Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs]: I removed these two lines from the file.

File: Taarafo.Core/Brokers/Storages/IStorageBroker.PostImpressions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cjdutoit) [reply Taarafo.Core/Brokers/Storages/IStorageBroker.PostImpressions.cs:15]: @evangelinedrink as you have multi-line method, you will either have to add an empty row before and after SelectPostImpressionByIdsAsync or if space permits move this to one line.

think you should be fine on one line though
```
        ValueTask<PostImpression> InsertPostImpressionAsync(PostImpression postImpression);
        ValueTask<PostImpression> SelectPostImpressionByIdsAsync(Guid PostId, Guid ProfileId);
        ValueTask<PostImpression> UpdatePostImpressionAsync(PostImpression postImpression);
```

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Please change to a single line

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs (hunk 1 above)
Comment (cjdutoit) [reply Taarafo.Core/Brokers/Storages/StorageBroker.PostImpressions.cs]: > Does EF know that this method needs to be queried based on these two `Ids`?

@ElbekDeveloper / @evangelinedrink, yes EF can find 

> I don't think the EF knows that the method needs to be queried based on the two Ids. How would I be able to verify this?

@evangelinedrink / @ElbekDeveloper EF can do a find for a composite key.  To give EF that info, we would need to add that fluent declaration to the `StorageBroker.PostImpressions.Configurations.cs` file as has been done already.

```
            modelBuilder.Entity<PostImpression>()
                   .HasKey(postImpression =>
                       new { postImpression.PostId, postImpression.ProfileId });
```
See this for more info on composite keys: [https://www.learnentityframeworkcore.com/configuration/fluent-api/haskey-method](https://www.learnentityframeworkcore.com/configuration/fluent-api/haskey-method)

**Comment by cjdutoit** [comment]
@evangelinedrink please resolve the branch conflicts as well

**Comment by hassanhabib** [comment]
@evangelinedrink friendly reminder.

**Pull request closed without merging**